File: src/main/java/org/team1/nbe1_3_team01/domain/group/service/response/BelongingResponse.java
package org.team1.nbe1_3_team01.domain.group.service.response;

import lombok.Builder;
import lombok.Data;
import org.team1.nbe1_3_team01.domain.group.entity.Belonging;

@Data
public class BelongingResponse {

    private Long id;
    private Long userId;
    private Long teamId;

    @Builder
    private BelongingResponse(Long id, Long userId, Long teamId) {
        this.id = id;
        this.userId = userId;
        this.teamId = teamId;
    }

    public static BelongingResponse from(Belonging belonging) {
        return BelongingResponse.builder()
                .id(belonging.getId())
                .userId(belonging.getUser() != null ? belonging.getUser().getId() : null)
                .teamId(belonging.getTeam() != null ? belonging.getTeam().getId() : null)
                .build();
    }

}

File: src/main/java/org/team1/nbe1_3_team01/domain/group/service/response/BelongingResponse.kt
package org.team1.nbe1_3_team01.domain.group.service.response

import org.team1.nbe1_3_team01.domain.group.entity.Belonging

data class BelongingResponse (
    val id: Long,
    val userId: Long,
    val teamId: Long
) {
    companion object {
        fun from(belonging: Belonging): BelongingResponse {
            return BelongingResponse(
                id = belonging.id!!,
                userId = belonging.user.id!!,
                teamId = belonging.team!!.id!!
            )
        }
    }
}
